src/internal/connector/graph/concurrency_middleware.go

```diff
@@ -0,0 +1,196 @@
+package graph
```

Comment:
nit: `concurrency_middleware.go` name is slightly misleading, given we have also added rate limiter to this file.  `limiter_middleware.go` maybe?

Reply:
`limiter` is equally generic.  What does it limit exactly?  Dinosaurs, perhaps...

Two-way door.  If one of us comes up with a great name, we can always change it up in a follow-up.